rmw_implementation/src/functions.cpp

```diff
@@ -46,11 +46,22 @@ std::string get_env_var(const char * env_var)
   return value ? value : "";
 }
 
-Poco::SharedLibrary *
+rcutils_shared_library_t *
 get_library()
 {
-  static Poco::SharedLibrary * lib = nullptr;
+  static rcutils_shared_library_t * lib = nullptr;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+
   if (!lib) {
+    lib = static_cast<rcutils_shared_library_t *>(allocator.allocate(
+            sizeof(rcutils_shared_library_t), allocator.state));
```

Comment:
Same kind of questions/comments as the other pr:

- weird indent
- shouldn't this be down where the `new` used to be?

Reply:
Also, make sure this is not leaked when this function fails in other cases below...

Reply:
used `new`